python/lsst/obs/base/camera_tests.py

```diff
@@ -0,0 +1,77 @@
+from __future__ import absolute_import, division, print_function
+from future.utils import with_metaclass
+#
+# LSST Data Management System
+# Copyright 2016 LSST Corporation.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+import abc
+import collections
+
+import lsst.afw.geom
+
+
+class CameraTests(with_metaclass(abc.ABCMeta)):
+    """
+    Tests that the butler returns a useable Camera.
+
+    In the subclasses's setUp():
+        * Instantiate a Butler on the testdata repository as self.butler.
+        * Call setUp_camera() to fill in required parameters.
+    """
+
+    def setUp_camera(self,
+                     camera_name='DECAM',
+                     n_detectors=62,
+                     first_detector_name='S29'
+                     ):
+        """
+        Set up the necessary variables for camera tests.
+
+         camera_name: str
+             name of this camera
+         n_detectors: int
+            number of detectors in this camera
+         first_detector_name: str
+            name of the first detector in this camera
+        """
+        fields = ['camera_name',
+                  'n_detectors',
+                  'first_detector_name'
+                  ]
+        CameraData = collections.namedtuple("CameraData", fields)
+        self.cameraData = CameraData(camera_name=camera_name,
+                                     n_detectors=n_detectors,
+                                     first_detector_name=first_detector_name
```

Comment:
I went with CameraData->cameraData because it behaves like a class/struct (which are camelcase in python). I'm not wedded to that choice, but it is a bit of an ambiguity in naming styles.